### DISCUSSION TITLE
How to reuse `opengraph-image.tsx` ?

### DISCUSSION THREAD

**Discussion by Falci** [vercel/next.js#51567] [App Router] [answered]
I have a complex `opengraph-image.tsx` under `/app/path-a/[name]` which I'd like to reuse on `/app/not-sibling/path-b/[name]` 

I tried: `export * from "./app/path-a/[name]/opengraph-image.tsx";` but no success.

What's the proper way to reuse it?

Edit: https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image#

**Comment by icyJoseph** [comment]
Mmm? You mean to do something like: `import { GET } from './app/path-a/[name]/opengraph-image.'`

**Reply by Falci** [reply]
Not really, [opengraph-image.tsx](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image#) exports some other const

**Comment by Falci** [comment, marked as answer]
Weird, but works:

```tsx
// /app/not-sibling/path-b/[name]/opengraph-image.tsx
import main from "app/path-a/[name]/opengraph-image";

// Route segment config
export const runtime = "edge";

export default main;
export * from "app/path-a/[name]/opengraph-image";
```

**Reply by icyJoseph** [reply]
Aha, mmm I had not used that opengraph-image file, Looks like you had to export the default and other stuff too. I guess your way is correct 😅

**Comment by christianonoh** [comment]
@Falci @icyJoseph  I have a similar but a little bit different. How do I actually access the generated image?
I have my `opengraph-image.tsx` under `/app/blog/[slug]` and I want to use this generated image as my jsonLd image in my  `/app/blog/[slug]/page.tsx` file but I dont know how to access this generated image.

**Comment by christianonoh** [comment]
@Falci @icyJoseph  I have a similar but a little bit different. How do I actually access the generated image?
I have my `opengraph-image.tsx` under `/app/blog/[slug]` and I want to use this generated image as my jsonLd image in my  `/app/blog/[slug]/page.tsx` file but I dont know how to access this generated image.

**Comment by christianonoh** [comment]
@Falci @icyJoseph  I have a similar but a little bit different. How do I actually access the generated image?
I have my `opengraph-image.tsx` under `/app/blog/[slug]` and I want to use this generated image as my jsonLd image in my  `/app/blog/[slug]/page.tsx` file but I dont know how to access this generated image.
